group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -517,15 +524,78 @@ public Assignment targetAssignment(String memberId) {
     }
 
     /**
-     * Updates target assignment of a member.
+     * @return An immutable map containing all the topic partitions
+     *         with their current member assignments.
+     */
+    public Map<Uuid, Map<Integer, String>> invertedTargetAssignment() {
+        return Collections.unmodifiableMap(invertedTargetAssignment);
+    }
+
+    /**
+     * Updates the target assignment of a member.
      *
      * @param memberId              The member id.
      * @param newTargetAssignment   The new target assignment.
      */
     public void updateTargetAssignment(String memberId, Assignment newTargetAssignment) {
```

Comment:
@rreddy-22 I was looking at the failed tests and I found a few which are related to your PR. I was curious so I took a deeper look into them and I found that we forgot to update the inverted target assignment in `removeTargetAssignment` too. Could you please take a look and fix this? We should also add more tests, I suppose.

Reply:
Ohno thanks!